catroid/src/main/java/org/catrobat/catroid/ui/adapter/BrickAdapter.java

```diff
@@ -1138,6 +1144,22 @@ private boolean brickHasAFormula(Brick brick) {
 		return (brick instanceof FormulaBrick || multiFormulaValid);
 	}
 
+	private boolean isBrickWithoutDescription(Brick brick) {
+		String name = brick.getClass().getSimpleName();
+		if (name.equals(IfLogicElseBrick.class.getSimpleName())) {
+			return true;
+		} else if (name.equals(IfLogicEndBrick.class.getSimpleName())) {
+			return true;
+		} else if (name.equals(IfThenLogicEndBrick.class.getSimpleName())) {
+			return true;
+		} else if (name.equals(LoopEndlessBrick.class.getSimpleName())) {
+			return true;
+		} else if (name.equals(LoopEndBrick.class.getSimpleName())) {
+			return true;
+		}
+		return false;
+	}
+
```

Comment:
Apply all the Android Studio inspection warnings (Yellow light bulbs).

Why do you compare the classes by `getSimpleName` instead of the `class` (`==`) itself or `instanceof`?

The return value of `isBrickWithoutDescription` is always used inverted. So invert the method name and adapt the code accordingly. `!isBrickWithoutDescription` -> `isBrickWithDescription`
Personally I prefer a name like `hasDescription(Brick brick)`.